src/rate-limit-view/rate-limit-view.ts

```diff
@@ -0,0 +1,89 @@
+import COLORS from "../data/rate-limit-view-colors.json";
+
+interface RateLimitColors {
+	negative: string,
+	positive: string[]
+}
+
+const RATE_LIMIT_COLORS: RateLimitColors = COLORS;
+const RATE_LIMIT = 333;
+const COLOR_INTERVAL = 20;
+const COLOR_MAX_MS = RATE_LIMIT_COLORS.positive.length * COLOR_INTERVAL;
+
+export class RateLimitView {
+	static oldtime = Date.now();
+	static newtime = Date.now();
+	static bgColor = "";
+	static fontColor = "";
+	static isDelta = false;
+	static isEnabled = false;
+
+	constructor(Options: any) {
+		$("<span id=ratelim class=ratelimit>RL: <span id=rate></span></span>").insertBefore(".CANT_REPLY");
+		$(".ratelimit").css({
```

Comment:
You should be able to reuse the previous reference here to the `#ratelim` element here.